### PR TITLE
Salt configuration to add basic auth to GCE.

### PR DESCRIPTION
**Pull request by roberthbailey**
This doesn't quite work yet -- kube-up succeeds but the script exits abruptly before cluster validation. I'm waiting for #7383 and #7425 to be merged so I can rebase this on top of them and figure out where in the shell needs to be tweaked. 

/cc @brendandburns

### CODE REVIEW

File: cluster/common.sh [hunk 1]
```diff
@@ -58,16 +60,17 @@ function create-kubeconfig() {
     )
   fi
   local user_args=()
-  if [[ -z "${KUBE_USER:-}" || -z "${KUBE_PASSWORD:-}" ]]; then
-    user_args+=(
-     "--token=${KUBE_BEARER_TOKEN}"
-    )
-  else
+  if [[ ! -z "${KUBE_USER:-}" && ! -z "${KUBE_PASSWORD:-}" ]]; then
```
Comment (roberthbailey) [inline cluster/common.sh]: @jlowdermilk - If I set --username, --password, and --token at the same time, does kubectl config set-credentials return an error?

Comment (jlowdermilk) [inline cluster/common.sh]: [Yes](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/cmd/config/create_authinfo.go#L207). You can only set username/password or token, and doing so will clear the other if it is present.

**Comment by zmerlynn** [comment]
I'd merge but Jenkins is hosed. :(

**Comment by roberthbailey** [comment]
@zmerlynn @jlowdermilk PTAL (still running e2e locally so don't merge yet).

**Comment by zmerlynn** [comment]
Still LGTM, just give me a green light.

**Comment by roberthbailey** [comment]
Ran 36 of 41 Specs in 1255.392 seconds
SUCCESS! -- 36 Passed | 0 Failed | 1 Pending | 4 Skipped I0428 14:54:36.866815   18977 driver.go:96] All tests pass

**Pull request merged**